app/services/creators/cash_transactions_creator.rb

```diff
@@ -63,7 +60,19 @@ def validate_category(category_hash)
 
     def validate_payment_dates(category_hash)
       dates = category_hash[:payments].map { |payment| Date.parse(payment[:date]) }.sort
-      @errors << "Expecting payment dates for category #{category_hash[:category]} to be #{pretty_dates}" unless dates == valid_dates
+      return if dates == first_three_valid_dates
+
+      return if dates == last_three_valid_dates
+
+      @errors << "Expecting payment dates for category #{category_hash[:category]} to be 1st of three of the previous 3 months"
+    end
+
+    def first_three_valid_dates
+      valid_dates.slice(0, 3)
```

Comment:
should it be `valid_dates.slice(0, 2)`
so it's the first three

Reply:
@Obsiye hiyah, no, the first argument stipulates the index to start from, whereas the second argument indicates the number of objects to slice from the array rather than the index number to slice to. It can look like `(start_index, end_index)` whereas it is `(start_index, total)`
e.g.
```
irb(main):040:0> arr = [0,1,2,3]
irb(main):041:0> arr.slice(2,1)
=> [2]
```